sys/riscv/riscv/pmap.c

```diff
@@ -529,15 +532,27 @@ pmap_bootstrap_dmap(vm_offset_t kern_l1, vm_paddr_t min_pa, vm_paddr_t max_pa)
 
 	/* Set the upper limit of the DMAP region */
 	dmap_phys_max = pa;
+
+#ifdef __CHERI_PURE_CAPABILITY__
+	dmap_capability = cheri_setaddress(cheri_kall_capability,
+	    DMAP_MIN_ADDRESS);
+	dmap_capability = cheri_setbounds(dmap_capability,
```

Comment:
Do we want to use `setboundsexact` here?

Reply:
I'm not sure we can.  All of RAM has to be accessible via the direct map, so we can't truncate the length (or if we did, we'd have to retroactively change all of phys_avail[] etc. to remove the RAM from all the VM's tables).  At the same time, there's no guarantee about how much RAM we will have.  Unlike MIPS where the associated capability is statically sized to the length of, e.g. the K0SEG in the address space whether there is RAM or not, this capability only tries to cover the portion of the direct map region that is backed by RAM and mapped via page tables rather than using the bounds of (DMAP_MIN_ADDRESS, DMAP_MAX_ADDRESS).  We could perhaps use exact bounds if we did the latter, but that could result in a capability with even broader bounds than the current code.